Classes/Application/SyncWorkspace/ConflictsBuilder.php

```diff
@@ -0,0 +1,294 @@
+            return null;
+        }
+
+        $dimensionSpacePoint = match (true) {
+            $command instanceof MoveNodeAggregate =>
+                $command->dimensionSpacePoint,
+            $command instanceof SetNodeProperties,
+            $command instanceof SetSerializedNodeProperties,
+            $command instanceof CreateNodeAggregateWithNode,
+            $command instanceof CreateNodeAggregateWithNodeAndSerializedProperties =>
+                $command->originDimensionSpacePoint->toDimensionSpacePoint(),
+            $command instanceof SetNodeReferences,
+            $command instanceof SetSerializedNodeReferences =>
+                $command->sourceOriginDimensionSpacePoint->toDimensionSpacePoint(),
+            $command instanceof TagSubtree,
+            $command instanceof DisableNodeAggregate,
+            $command instanceof UntagSubtree,
+            $command instanceof EnableNodeAggregate,
+            $command instanceof RemoveNodeAggregate =>
+                $command->coveredDimensionSpacePoint,
+            $command instanceof ChangeNodeAggregateType =>
+                null,
+            $command instanceof CreateNodeVariant =>
+                $command->targetOrigin->toDimensionSpacePoint(),
+            default => null
+        };
+
+        if ($dimensionSpacePoint === null) {
+            if ($nodeAggregateIdForDimensionFallback === null) {
+                return null;
+            }
+
+            $nodeAggregate = $this->contentRepository->getContentGraph()
+                ->findNodeAggregateById(
+                    $this->workspace->currentContentStreamId,
+                    $nodeAggregateIdForDimensionFallback
+                );
+
+            if ($nodeAggregate) {
+                $dimensionSpacePoint = $this->extractValidDimensionSpacePointFromNodeAggregate(
+                    $nodeAggregate
+                );
+
+                if ($dimensionSpacePoint === null) {
+                    return null;
+                }
+            }
+        }
+
+        return $this->contentRepository->getContentGraph()->getSubgraph(
+            $this->workspace->currentContentStreamId,
+            $dimensionSpacePoint,
+            VisibilityConstraints::withoutRestrictions()
+        );
+    }
+
+    private function extractValidDimensionSpacePointFromNodeAggregate(
+        NodeAggregate $nodeAggregate
+    ): ?DimensionSpacePoint {
+        $result = null;
+
+        foreach ($nodeAggregate->coveredDimensionSpacePoints as $coveredDimensionSpacePoint) {
+            if ($this->preferredDimensionSpacePoint?->equals($coveredDimensionSpacePoint)) {
+                return $coveredDimensionSpacePoint;
+            }
+            $result ??= $coveredDimensionSpacePoint;
+        }
+
+        return $result;
+    }
+
+    private function createIconLabelForNode(Node $node): IconLabel
+    {
+        return new IconLabel(
+            icon: $node->nodeType?->getConfiguration('ui.icon') ?? 'questionmark',
```

Comment:
As we want to get rid of Node::nodeType and we have the ContentRepository available, we can already fetch the nodeType from there

Reply:
Very good point. Even if we skip NodeTypeManager overhaul for 9.0 I would really like to get rid of Node.nodeType because that will allow us to serialize the nodes (e.g. allowing us to solve the "higher level hook" to use a job queue)

Reply:
Adjusted :)